src/js/view/controls/components/menu/settings-menu.js

```diff
@@ -27,6 +27,7 @@ const SettingsMenu = (api, model, controlbar, localization) => {
             controlBarButton.show();
         }
     };
+    let levelChangedManually = false;
```

Comment:
is this leftover code? i don't see it in use beyond being set tot `true/false`

Reply:
When you change the level manually, the api is set to dispatch a 'levels' event, which in this case will cause an unnecessary re-render of the quality menu (items didn't change, simply selected one, so better not to re-render). Rob walked me through why that event dispatch was necessary, so this was my way of preventing the additional reflow when that happens.

Reply:
Only with the html5 provider and multiple mo4 sources which is a really small percentages of embeds. Any provider could dispatch "levels" for it's own reasons. I don't think fitting this flag to that one edge case makes sense. This optimization for such a small usecase (mp4 levels) is not worth the extra code. It would be better just always redraw the menu.

Reply:
This was the least lines of code I could come up with to achieve the desired result. Otherwise we run into this issue:

I click the item -> It sets the level -> Before anything else executes, the 'levels' event is dispatched -> The menu is recreated and the items are replaced with the new items -> the old items function continues, and garbage isn't collected until the function is finished -> any item transformations are done on the orphaned nodes before they're garbage collected

Now I can add state to the items to fix this, but this was the simplest way I came up with to prevent this in an OOP pattern.

Reply:
Actually, just came up with something. I'll try it out and report back.